### PR TITLE
Update provider.ts

### PR DESCRIPTION
**Pull request by mxdh**
Solve path problems.

### CODE REVIEW

**Comment by dseight** [comment]
I suppose that we should not leave path unresolved. Instead, we may do something like this:
```patch
diff --git a/src/provider.ts b/src/provider.ts
index 731a93c..0fa4a55 100644
--- a/src/provider.ts
+++ b/src/provider.ts
@@ -83,7 +83,7 @@ export function encodeAsmUri(uri: Uri): Uri {
             let associated = associations[key];
             return uri.with({
                 scheme: AsmProvider.scheme,
-                path: resolvePath(uri.path, associated)
+                path: resolvePath(uri.fsPath, associated)
             });
         }
     }
```
`uri.fsPath` will be a path without leading slash on Windows. Also, it will use system-specific path delimiters (backslashes on Windows and regular slashes on Linux/macOS).

Probably, we need to get `parsedWorkspacePath` in the same way:
```patch
diff --git a/src/provider.ts b/src/provider.ts
index 731a93c..05d877c 100644
--- a/src/provider.ts
+++ b/src/provider.ts
@@ -99,7 +99,7 @@ function resolvePath(path: string, associated: string): string {
     }
 
     let parsedFilePath = Path.parse(path);
-    let parsedWorkspacePath = Path.parse(workspace.workspaceFolders[0].uri.path);
+    let parsedWorkspacePath = Path.parse(workspace.workspaceFolders[0].uri.fsPath);
 
     let variables: any = {
         // the path of the folder opened in VS Code
```

**Comment by mxdh** [comment]
Sorry to have bothered you.
I am a novice and I am not very familiar with javascript or typescript.
Thank you.

**Comment by dseight** [comment]
You can just update your merge request (e.g. by force pushing to your branch) with supposed changes (if you wish) and I will merge it 😉

File: src/provider.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ export class AsmProvider implements TextDocumentContentProvider {
     }
 
     reloadAsmDocument(fileUri: Uri) {
-        const uri = fileUri.with({scheme: AsmProvider.scheme});
+        const uri = fileUri.with({ scheme: AsmProvider.scheme });
```
Comment (dseight) [inline src/provider.ts:41]: Note that it is always better to have style-related changes in a separate commit (as they are don't do anything to really fix the issue).

Comment (mxdh) [reply src/provider.ts:41]: Thank you for your reminding.
It may be caused by VS Code's auto format.

**Pull request merged**